### PR TITLE
Custom Icon

### LINKED ISSUE #327: Custom Icon
### Component in Question
- `Icon`

### Description of Issue
- You should be able to [pass in a url](https://nypl.slack.com/archives/CLH9PL04E/p1596727465340600) to use the icon sizes
- Story should include information that of it's broken you check the viewPort of your SVG

### PR DESCRIPTION
**Pull request by helenvholmes**
Issue number: #327

## **This PR does the following:**
- Allows consumers to pass in a custom icon as children to `Icon`

### Front End Review:
- [ ] View [the example in Storybook](https://deploy-preview-349--stoic-murdock-c7f044.netlify.app/?path=/story/icon--custom-icon)

### CODE REVIEW

File: src/components/Icons/Icon.tsx [hunk 1]
```diff
@@ -55,51 +55,56 @@ export interface IconProps {
     /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
     modifiers?: string[];
     /** Name of the icon */
-    name: IconNames | LogoNames;
+    name?: IconNames | LogoNames;
```
Comment (hokei) [inline src/components/Icons/Icon.tsx:58]: What happens when you don't pass a name or a child?

File: src/components/Icons/Icon.tsx [hunk 2]
```diff
@@ -55,51 +55,56 @@ export interface IconProps {
...
-            "aria-hidden": decorative,
-            "aria-labelledby": title ? "title-" + name : undefined,
-            "aria-describedby": desc ? "desc-" + name : undefined,
-            title: title ? `title-${name}` : undefined,
-            description: desc ? `title-${name}` : undefined,
-        };
+    let svg;
 
-        let svg = allSvgs[name];
+    if (name && children) {
```
Comment (hokei) [inline src/components/Icons/Icon.tsx:97]: let's add a test for this

File: src/components/Icons/Icon.tsx (hunk 1 above)
Comment (hokei) [inline src/components/Icons/Icon.tsx:58]: What happens when you pass neither a name nor a child?

Comment (helenvholmes) [reply src/components/Icons/Icon.tsx:58]: Good point. I added a warning

File: src/components/Icons/Icon.tsx (hunk 2 above)
Comment (M0nica) [reply src/components/Icons/Icon.tsx:97]: +1

Comment (helenvholmes) [reply src/components/Icons/Icon.tsx:97]: 💯

**Review by hokei — APPROVED** [review]

**Review by M0nica — APPROVED** [review]

**Pull request merged**